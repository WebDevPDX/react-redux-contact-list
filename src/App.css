.app-wrapper {
  display: flex;
  margin: 0;
  padding: 10px;
  height:100vh;
  justify-content: center;
  background-color: midnightblue;
}
