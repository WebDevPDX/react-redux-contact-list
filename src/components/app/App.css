.app-wrapper {
  display: flex;
  margin: 0;
  padding: 10px;
  height: 100vh;
  max-width: 640px;
}
