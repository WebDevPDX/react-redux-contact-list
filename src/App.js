import React, { Component } from 'react'
import { BrowserRouter as Router, Route } from "react-router-dom"
import { ContactsList } from './components/contactsList/ContactsList'
import DetailContact from './containers/detailContact/DetailContact'
import AddContactForm from './components/addContactForm/AddContactForm'
import { UserForm } from './components/userForm/UserForm'
import './App.css';

class App extends Component {
  render() {
    return (
      <div className='app-wrapper'>
        <Router>
          <Route path="/" exact component={ContactsList} /> 
          <Route path="/user/:id" component={DetailContact} />
          <Route path="/edit/:id" component={UserForm} />
          <Route path="/add" component={AddContactForm} />
        </Router>
      </div>
    )
  }
}

export default App;
