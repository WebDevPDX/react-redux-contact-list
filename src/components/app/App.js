import React, { Component } from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import AllContactsContainer from '../../containers/allContacts/AllContactsContainer'
import DetailedContactContainer from '../../containers/detailedContact/DetailedContactContainer'
import ContactFormContainer from '../../containers/contactFormContainer/ContactFormContainer'
import './App.css'

class App extends Component {
  render() {
    return (
      <div className="app-wrapper">
        <Router>
          <Switch>
            <Route path="/" exact component={AllContactsContainer} />
            <Route path="/user/:id" exact component={DetailedContactContainer} />
            <Route path="/user/edit/:id" exact component={ContactFormContainer} />
            <Route path="/add" component={ContactFormContainer} />
            <Route component={AllContactsContainer} />
          </Switch>
        </Router>
      </div>
    )
  }
}

export default App
