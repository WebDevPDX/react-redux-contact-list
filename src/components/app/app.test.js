import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import App from './App'

it('renders correctly', () => {
  const renderer = new ShallowRenderer()
  const tree = renderer
    .render(<App />)

  expect(tree).toMatchSnapshot();
});