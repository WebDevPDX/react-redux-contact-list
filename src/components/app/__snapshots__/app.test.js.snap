// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app-wrapper"
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/user/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/user/edit/:id"
      />
      <Route
        component={[Function]}
        path="/add"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
